runtime-modules/referendum/src/lib.rs

```diff
@@ -827,20 +828,25 @@ impl<T: Trait<I>, I: Instance> EnsureChecks<T, I> {
     }
 
     fn can_release_vote_stake(origin: T::Origin) -> Result<T::AccountId, Error<T, I>> {
-        let cycle_id = CurrentCycleId::<I>::get();
-
         // ensure superuser requested action
         let account_id = Self::ensure_regular_user(origin)?;
 
         let cast_vote = Self::ensure_vote_exists(&account_id)?;
 
+        // get current cycle id
+        let current_cycle_id = match Stage::<T, I>::get() {
```

Comment:
I guess the question is here: should we do this check here, or just leave it to the client to decide whether its sometimes ok to even unstake in the same election cycle? If it happens during voting, then there is no harm really, but if the client allows it during revealing, the the client could allow people to unstake regardless of what the relationship is between the vote and the outcome. I think that is fine in principle, in some voting systems it could be perfectly acceptable.

A benefit of keeping all the checks in the council is that you have one small place that holds all the rules, rather than splitting it across the two modules like this.

Reply:
The conditioning has been moved to council.